### PR TITLE
Parallelize ShapeIt4 over genome regions

### PR DESCRIPTION
**Pull request by jessicaway**
ShapeIt4 was taking a very long time to run, this splits over genome regions and ligates

### CODE REVIEW

**Review by alimanfoo — APPROVED** [review]
LGTM, thanks @jessicaway.

File: scripts/create_interval_lists.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+
+"""
+Creates interval lists in the specified output directory for the chromosomes listed in the chromosome lenghts file.
+An example of a chromosome lengths file is at:
+gs://malariagen/references/Anopheles-gambiae-PEST_CHROMOSOMES_AgamP4/Anopheles-gambiae-PEST_CHROMOSOMES_AgamP4.chr_lengths.txt
+Default interval size is 2Mb and default overlap size is 400Kb (note that arguments to the script are in bp).
+"""
+def main():
+    parser = argparse.ArgumentParser(description="Creat an interval list for each chromosome")
```
Comment (gbggrant) [inline scripts/create_interval_lists.py:51]:
```suggestion
    parser = argparse.ArgumentParser(description="Create an interval list for each chromosome")
```

File: tasks/gcp/StatisticalPhasingTasks.wdl [hunk 2]
```diff
@@ -85,11 +87,76 @@ task ShapeIt4 {
...
+
+  command {
+    set -e pipefail
+
+    python3 <<CODE
+    from pathlib import Path
+
+    index_files = [ "~{sep='", "' region_phased_vcfs_indices}" ]
+    for f in index_files:
+        Path(f).touch()
```
Comment (gbggrant) [inline tasks/gcp/StatisticalPhasingTasks.wdl:120]: Do you still need this touch?

File: tasks/gcp/StatisticalPhasingTasks.wdl [hunk 3]
```diff
@@ -85,11 +87,76 @@ task ShapeIt4 {
...
+
+    bgzip ~{project_id}_phased.vcf
+    tabix ~{project_id}_phased.vcf.gz
+  }
+
+  runtime {
+    docker: docker_tag
+    preemptible: preemptible_tries
+    cpu: num_cpu
+    memory: "15 GiB"
```
Comment (gbggrant) [inline tasks/gcp/StatisticalPhasingTasks.wdl:149]: Should this be a task-level argument?

**Review by gbggrant — APPROVED** [review]

File: tasks/gcp/StatisticalPhasingTasks.wdl (hunk 3 above)
Comment (jessicaway) [reply tasks/gcp/StatisticalPhasingTasks.wdl]: good idea, I have made memory a task level argument for all tasks in this wdl

File: tasks/gcp/StatisticalPhasingTasks.wdl (hunk 2 above)
Comment (jessicaway) [reply tasks/gcp/StatisticalPhasingTasks.wdl:123]: Probably don't need this, but it's working and not giving warnings, so I am going to leave it for now

**Pull request merged**